src/bin/jp2/opj_compress.c

```diff
@@ -486,6 +486,9 @@ static unsigned int get_num_images(char *imgdirpath)
             continue;
         }
         num_images++;
+        if (num_images == 0) {
```

Comment:
while this is technically correct, having behaviour that relies on overflow of unsigned types always makes me a bit nervous. I'd prefer a test on UINT_MAX before the incrementation is done (the advantage is that a build with  clang -fsanitize=undefined,unsigned-integer-overflow would also be happy). Same for below similar instances

Reply:
Revised as requested.